app/main/services/query_builder.py

```diff
@@ -11,6 +13,101 @@
     "supplierName"
 ]
 
+FILTER_FIELDS = [
+    "lot",
+    "serviceTypes",
+    "freeOption",
+    "trialOption",
+    "minimumContractPeriod",
+    "supportForThirdParties",
+    "selfServiceProvisioning",
+    "datacentresEUCode",
+    "dataBackupRecovery",
+    "dataExtractionRemoval",
+    "networksConnected",
+    "apiAccess",
+    "openStandardsSupported",
+    "openSource",
+    "persistentStorage",
+    "guaranteedResources",
+    "elasticCloud"
+]
+
+
+class QueryFilter(object):
+    """
+    Class to encapsulate a query multidict from flask
+
+    Filter name is the key from the multidict.
+
+    Multidict is map of key to a list, to encompass
+    multiple HTTP params of same name
+
+    Filter value is derived from the mutlidict
+        - Single string value is an AND filter
+        - Single string value, that contains commas is
+        treated as a CSV and becomes
+        an OR filter with each string a single term
+        - Multiple values is treated as an AND with
+        several terms for that key
+
+    Examples:
+        filter_lot=saas&filter_lot=paas => lot == saas AND lot = paas
+        filter_lot=saas,paas => lot == saas OR lot ==saas
+        filter_lot=saas => lot == saas
+
+
+    """
+    OR = "or"
+    AND = "and"
+
+    def __init__(self, field, values):
+        self.filter_field = field
+        self.filter_values = values
+        self.filter_type = self.__filter_type()
+
+    def __filter_type(self):
+        # multiple values for same key is an AND filter
+        if len(self.filter_values) > 1:
+            return self.AND
+        if len(self.filter_values) == 1:
+            if re.search(",", self.filter_values[0]):
```

Comment:
Some of the data values allowed in our schemas do have commas. none that we use (currently) in filters though.

By no means welded to commas though - can change the character, but more probably a JSONSchema extension or some means of inference may be better. Thing is do we want the buyer app, or other clients of search, do be able to ask AND or OR, or do we want that to be part of the framework. 

Thinking the former - so a new character may be useful if commas become an issue. Note this should be encapsulated in the client so should be a "not to hard" migration.